spectral_cube/stokes_spectral_cube.py

```diff
@@ -45,31 +67,31 @@ def __init__(self, stokes_data, mask=None, meta=None, fill_value=None):
                 raise ValueError("All spectral cubes in stokes_data "
                                  "should have the same WCS")
 
-            if component not in VALID_STOKES:
-                raise ValueError("Invalid Stokes component: {0} - should be one of I, Q, U, V, RR, LL, RL, LR, XX, XY, YX, YY, \
-                                 RX, RY, LX, LY, XR, XL, YR, YL, PP, PQ, QP, QQ, \
-                                 RCircular, LCircular, Linear, Ptotal, Plinear, PFtotal, PFlinear, Pangle".format(component))
+            # Validation is now handled by StokesCoord above
 
             if stokes_data[component].shape != stokes_data[reference].shape:
                 raise ValueError("All spectral cubes should have the same shape")
 
         self._wcs = stokes_data[reference].wcs
         self._shape = stokes_data[reference].shape
+        self._stokes_coord = stokes_coord  # Store StokesCoord instance
```

Comment:
```suggestion
        self._stokes_coord = stokes_coord
```
the comment is redundant